keps/sig-instrumentation/0034-distributed-tracing-kep.md

```diff
@@ -0,0 +1,299 @@
+
+While adding tracing to Pods is a good first step to demonstrate the viability of object lifecycle tracing in kubernetes, we expect component owners to add tracing to their components in an ad-hoc fashion.
+
+### Out-of-tree changes
+
+#### Tracing best-practices documentation
+
+This KEP introduces a new form of instrumentation to Kubernetes, which necessiates the creation of guidelines for adding effective, standardized traces to Kubernetes components, [similar to what is found here for metrics](https://github.com/kubernetes/community/blob/master/contributors/devel/instrumentation.md).
+
+This documentation will put forward standards for: 
+
+* How to name spans, attributes, and annotations
+* What kind of processes should be wrapped in a span
+* When to link spans to other traces
+* What kind of data makes for useful attributes
+* How to propagate trace context as proposed above
+
+Having these standards in place will ensure that our tracing instrumentation works well with all backends, and that reviewers have concrete criteria to cross-check PRs against. 
+
+## Graduation requirements
+
+Alpha
+
+- [] Alpha-implementation as described above
+- [] E2e testing of traces
+- [] User-facing documentation
+- [] Tracing must not increase the number of requests to the APIServer
+
+Beta
+
+- [] Security Review, including threat model
+- [] Deployment review including whether the [OC Agent](https://github.com/census-instrumentation/opencensus-service#opencensus-agent) is a required component
+- [] Benchmark kubernetes components using tracing, and determine resource requirements and scaling for any additional required components (e.g. OC Agent).
+
+GA
+
+- [] Versioning for span naming and backwards-compatibility guarantees
+
+## Production Readiness Survey
+
+* Feature enablement and rollback
+  - How can this feature be enabled / disabled in a live cluster?  **Feature-gate**
+  - Can the feature be disabled once it has been enabled (i.e., can we roll
+    back the enablement)?  **Yes, the feature gate can be disabled in all relevant components**
+  - Will enabling / disabling the feature require downtime for the control
+    plane?  **Yes, control-plane components must be restarted with the feature-gate disabled.**
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?  **No, it just requires restarting the kubelet with the feature-gate disabled**
+  - What happens if a cluster with this feature enabled is rolled back? **No Adverse effects** What
+    happens if it is subsequently upgraded again?  **No adverse effects**
+  - Are there tests for this?  **No.  The feature hasn't been developed yet.**
+* Scalability
+  - Will enabling / using the feature result in any new API calls? **No (there was a recent change in the KEP to that effect).**
```

Comment:
nit: maybe that's only me (feel free to disagree), but i think it would be easier to read if the answer would be at the end of the question - now it's kind-of in the middle :)

Reply:
Done.